### ISSUE TITLE
Progress output (in stdout) is flushed only at the end of the run when redirected into a file

### ISSUE THREAD

**Issue by gdurif** [diyabc/diyabc#58]
When running either `diyabc-RF-linux-v1.1.24 -p ./ -R ""  -m -g 12 -r 1000 -t 6 > tmp.log` or `diyabc-RF-linux-v1.1.24 -p ./ -R ""  -m -g 12 -r 1000 -t 6 > tmp.log 2>&1`, all output below the following lines

```
----------------------------------------apres readHeaderAllStat
----------------------------------------apres buildMutParam
fin de readheader  nloc retenus=500
Renaming headerRFRF.txt to headerRF.txt
```
is not printed in the output, missing 

```
calstatobs (./statobsRF.txt)
stat_obs[0]=0.208
stat_obs[1]=0.268
stat_obs[2]=0.228
stat_obs[3]=0.174
stat_obs[4]=0.07
   .../...
stat_obs[124]=-0.00247344
stat_obs[125]=-0.00143472
stat_obs[126]=0.00103872
stat_obs[127]=0.000234732
stat_obs[128]=0.000313357
stat_obs[129]=0.000212723

verification de l'integrite de la table de reference 
fichier./reftableRF.bin
nrec = 120
nrecscen[0] = 18
nrecscen[1] = 23
nrecscen[2] = 22
nrecscen[3] = 19
nrecscen[4] = 17
nrecscen[5] = 21
nparam[0] = 8
nparam[1] = 8
nparam[2] = 8
nparam[3] = 7
nparam[4] = 7
nparam[5] = 7
nstat = 130
6 scenario(s)
150
fichier reftable OK

dans rt.readheader this->filerefscen = ./reftabscen.txt
nparam[0] = 8
nparam[1] = 8
nparam[2] = 8
nparam[3] = 7
nparam[4] = 7
nparam[5] = 7
debut de sethistparamname
nparamut=0    nscenarios=6
avant la boucle des scenarios  nscenarios =6
scenario 0   scenario[i].nparam=8  nparamvar=8
scenario 1   scenario[i].nparam=8  nparamvar=8
scenario 2   scenario[i].nparam=8  nparamvar=8
scenario 3   scenario[i].nparam=7  nparamvar=7
scenario 4   scenario[i].nparam=7  nparamvar=7
scenario 5   scenario[i].nparam=7  nparamvar=7
fin de sethistparamname
apres readheader   k=0

 nloc = 500   nsample = 4   k=0
DEBUT  nrecneeded=1000   rt.nrec=120    rt.nstat=130   nscenarios=6

Ouverture du fichier
./reftableRF.bin

nparammax=16
n simu data = 170
```
and in particular the progress monitoring `n simu data = XX`.

Without redirection (i.e. `diyabc-RF-linux-v1.1.24 -p ./ -R "" -m -g 12 -r 1000 -t 6`), no problem, all output appear in the terminal.

**Comment by gdurif** [comment]
To give more details, the "missing" output is written only at the end of the run (when redirected into a file) and not during the run, which is not practical to follow the simulation progression.

**Comment by gdurif** [comment]
Here are some tests:

- a bash script `test.sh`:
```bash
#!/bin/bash
for number in {1..1000}
do
    echo "simu = $number"
    sleep 30s
done
exit 0
```

The output of the command `./test.sh > tmp.log` run in a terminal is not buffered.

In R with `processx` to  run a detached process, the output is also not buffered and written into the log file on the fly:
```R
proc <- processx::process$new(
    command = "test.sh", 
    args = character(),
    stdin = NULL,
    stdout = "test.log", 
    stderr = "test.log",
    echo_cmd = TRUE
)
```

To control the detached processus:
```R
proc$is_alive()
proc$kill()

proc$get_exit_status()
```

- a C++ program (file `test.cpp` compiled into `test.bin` with `g++ test.cpp -o test.bin -std=c++11`):
```C++
#include <iostream>
#include <thread>  

int main()
{
    for(auto i = 1; i < 1000; i++) {
        std::cout << "simu = " << i << std::endl;
        std::this_thread::sleep_for (std::chrono::seconds(1));        
    }
}
```

`./test.bin > tmp.log` in ta erminal is not buffered, and same from R with:
```R
proc <- processx::process$new(
    command = "test.bin", 
    args = character(),
    stdin = NULL,
    stdout = "test.log", 
    stderr = "test.log",
    echo_cmd = TRUE
)
```

**Comment by fradav** [comment]
fixed in b15101444a5247563d3bc4208b408d7841bacfba
Release : https://github.com/diyabc/diyabc/releases/tag/v1.1.26
